CTP-5101 Fix intro width with Boost

leonstr opened the pull request:
Add max-width to intro section so this aligns with submissions table when using theme_boost. Also, placed warning messages for the main coursework page in a containing <div> to fix the same problem with these.

Also, changed warnings->alert_div() to use core's notification.

leonstr committed:
CTP-5101 Fix intro width with Boost

Add max-width to intro section so this aligns with submissions table
when using theme_boost. Also, placed warning messages for the main
coursework page in a containing <div> to fix the same problem with
these.

Also, changed warnings->alert_div() to use core's notification.

stuartlamour commented on templates/intro.mustache:
```diff
@@ -31,19 +31,29 @@
-    {{#generalfeedback}}
-        {{> mod_coursework/general_feedback}}
-    {{/generalfeedback}}
+<div class='row'>
```
yes! sorry i didn't do this initially, was on my radar...

stuartlamour commented on renderers/object_renderer.php:
```diff
@@ -436,7 +436,7 @@ public function render_plagiarism_links($files) {
         global $USER;
 
-        $out = '';
+        $out = '<div class="container">';
```
stick this in the template too?

stuartlamour committed:
Move more html to template, clean up php

leonstr committed:
CTP-5101 Move div into template

Also fix Not Enough Assessors warning in intro.

TODO Containing dif for teacher_grading_page()'s warnings.

leonstr committed:
CTP-5101 Optional items before submissions in div

Move optional items before the submissions table into <div
class="container">.

stuartlamour commented on renderers/page_renderer.php:
```diff
@@ -533,41 +533,36 @@ public function teacher_grading_page($coursework, $page, $perpage, $sortby, $sor
-            $html .= html_writer::start_div('mdl-right');
-            $html .= html_writer::link($url, get_string('resettable'));
-            $html .= html_writer::end_div();
+            $template->resettableurl = new moodle_url('/mod/coursework/view.php', ['id' => $coursework->get_course_module()->id, 'treset' => 1]);
```
I cannot see what this does.
The if ($firstnamealpha || $lastnamealpha || $groupnamealpha) sugests its from when datatables passed through sorting vars in the url.
probably need to get rid of all instances of treset and the other vars...

leonstr replied:
Added [CTP-5136 Coursework UX - Remove redundant filter code](https://ucldata.atlassian.net/browse/CTP-5136) to cover this.

stuartlamour commented on renderers/page_renderer.php:
```diff
@@ -533,41 +533,36 @@ public function teacher_grading_page($coursework, $page, $perpage, $sortby, $sor
+            $template->resettableurl = new moodle_url('/mod/coursework/view.php', ['id' => $coursework->get_course_module()->id, 'treset' => 1]);
         }
 
         if ($firstnamealpha || $lastnamealpha || $groupnamealpha || $group != -1) {
```
i think these are from datatables.
Probably need to remove all refs to $firstnamealpha || $lastnamealpha || $groupnamealpha  and associated.

leonstr replied:
Added [CTP-5136 Coursework UX - Remove redundant filter code](https://ucldata.atlassian.net/browse/CTP-5136) to cover this.

stuartlamour commented on templates/submissions/table.mustache:
```diff
@@ -1,29 +1,30 @@
-<div class="container py-3" id="mod-coursework-submissions-table-container-{{coursework.id}}">
-    <h3>{{#str}}submissions, mod_coursework{{/str}}</h3>
+<div class='wrapper_table_submissions'>
```
i'm not sure what this class does?
nothing in css, and the fact its using _ seems to indicate its probably legacy?
